### ISSUE TITLE
processing GEO data files

### ISSUE THREAD

**Issue by SLeviyang** [Manel-Lab/MVA_DC_scRNAseq#1]
I would like to use the scRNAseq files posted on GEO, GSE153835.  However, I'm having some trouble understanding where these files fit in the R downstream analysis posted here on github.  I see 4 files corresponding to each sample.  For example, for sample 1 - infected, I see

1.  GSM4655140_filtered_barcodes_D1689_MVA_05.tsv
2. GSM4655140_filtered_genes_D1689_MVA_05.tsv
3. GSM4655140_filtered_matrix_D1689_MVA_05.mtx
4. GSM4655140_raw_matrix_D1689_MVA_05.mtx

The filtered barcodes number 1301 which matches the CellRanger count in Figure S16C.  But then looking at the github code, I see a call to cellCalling in mvadcs_downstream.R which seems to convert raw to filtered, while using EmptyDrops.  So I'm confused on what code is used to generate the filtered files above and what code is used to generate the raw matrix.  Also, there is a metadata file posted on GEO (GSE153835_metadata.tsv).  What is this file?

Generally, I would like to recreate the dataset you used in the article.  But it seems that the filtered data on GEO is not quite that, and I'm uncertain which portions of the R code you applied to the data.  

Thanks,
Sivan

**Comment by kdeazevedo** [comment]
Hello !

To generate the files under each sample on GEO (barcode file, gene file, and filtered & raw matrix), **only** CellRanger was used. If you wanted to recreate these files, you'd have to follow instruction 1 to 3 of the README, creating a reference transcriptome of MVA/mcherry/human, and aligning the FASTQ on this transcriptome.

Once you have these files (or just use the ones in GEO if you want), the next step is indeed the downstream analysis, which starts with the cellCalling function. Basically, using the raw and filtered matrix from CellRanger, we **apply another round of filtering with EmptyDrops** to improve the quality of cells, and also try and save cells that would be filtered out by CellRanger because of low human RNA count, but that we would still like to keep if they have high viral RNA count. Once it's done, a Seurat object is created that will be used for the rest of the downstream analysis.

This Seurat object is not provided on GEO per se, but the raw and normalized counts resulting from the additional filtering with EmptyDrops are provided in tsv files (counts.tsv and normalized.tsv), and making a Seurat object from these should be possible I believe. The metadata.tsv file is a file that also describes the EmptyDrops-filtered dataset, with additional information resulting from clustering. Basically, it exists for people who wouldn't want to do the analysis from the beginning, and want the basic information from the analysis in text format, so they can play around with it if they want.

So basically, it depends on where your interest lies. If you just want the dataset to explore it, the tsv files should be good enough to then make a Seurat object that would be manipulable.

I hope it was clear enough, don't hesitate if you have further questions !
Kévin

**Comment by SLeviyang** [comment]
Thanks, I'm in the process of running parts of the pipeline.  One question, what does the parameter normalize do?  It seems that it is set to TRUE?  Is that the case for all analysis?

**Comment by kdeazevedo** [comment]
Hi,

As explained in Marius et. al (2021), we separately normalized human and viral features, as we found it performed better when integrating the data. The argument "normalize" is about this. TRUE would mean you would normalize features separately, FALSE would be a more standard normalization with all features normalized together. There is no choice to change the normalization method in itself, in both cases it is the LogNormalize normalization implemented in Seurat.
In our analysis, we set the argument to TRUE.

Kévin
